libs/wingc/src/diagnostic.rs

```diff
@@ -187,9 +187,16 @@ impl PartialOrd for WingSpan {
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
 pub struct Diagnostic {
 	pub message: String,
+	pub annotations: Vec<DiagnosticAnnotation>,
 	pub span: Option<WingSpan>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
+pub struct DiagnosticAnnotation {
+	pub message: String,
+	pub span: WingSpan,
+}
```

Comment:
Should we have include the concept of hints in this model? Obviously you can put it in the message, but what if it were a separate field or something here. Might encourage others to add hints, and maybe they can be formatted differently.

Also, a possible future for this: Would be interesting to integrate hints with LSP code actions to fix errors. That way these fixes can be in the compiler itself and not rewritten for the language server.

Reply:
Sounds like a good future improvement